clusterloader2/pkg/prometheus/experimental.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prometheus
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"os/exec"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+)
+
+var (
+	shouldSnapshotPrometheusDisk = pflag.Bool("experimental-gcp-snapshot-prometheus-disk", false, "(experimental, provider=gce|gke only) whether to snapshot Prometheus disk before Prometheus stack is torn down")
```

Comment:
This is something that I mentioned to you at some point in the past:
https://github.com/kubernetes/enhancements/issues/177

Can you please maybe ping the author directly over email and ask about the status of it whether it's something we can use here? If that already exists, I would prefer using something existing, especially that it won't introduce restrictions.

Reply:
The feature is currently in alpha, was supposed to be in alpha in 1.14, but didn't get in.

As discussed personally, leaving it as is for now. It's a hacky, but we don't really have capacity to do anything better now and it should do as a temporary solution.
In future we should investigate other options and rewrite this to something better, e.g. exporting prometheus data to some db or maybe there is already existing tool that we could use that addresses our requirements. 
Opened an issue to track this - https://github.com/kubernetes/perf-tests/issues/523